### PR TITLE
Better unicode.ml generator

### PR DESCRIPTION
**Pull request by hhugo**
- no longer extract information from comments
- extract missing information from UnicodeData.txt
- track ignored properties so that one can codereview them
- make the parsing of Unicode data fail if lines don't have the expected format.

Based on #144

### CODE REVIEW

**Comment by hhugo** [comment]
I've extracted some fix in #144

**Comment by hhugo** [comment]
This is now based on top of #144.
Only the last commit is part of this PR.
once #144 is merged, this PR will only change how unicode.ml is generated but will no change its content.

File: src/generator/gen_unicode.ml [hunk 1]
```diff
@@ -177,9 +145,58 @@ let () =
...
+      "po";
+      "ps";
+      "sc";
+      "sk";
+      "sm";
+      "so";
+      "zl";
+      "zp";
+      "zs";
+    ];
```
Comment (toots) [inline src/generator/gen_unicode.ml:180]: I would make this a constant to make it clear what lists we're considering.

File: src/generator/gen_unicode.ml [hunk 2]
```diff
@@ -177,9 +145,58 @@ let () =
...
+      "math";
+      "other_alphabetic";
+      "other_lowercase";
+      "other_math";
+      "other_uppercase";
+      "uppercase";
+      "white_space";
+      "xid_continue";
+      "xid_start";
+    ];
```
Comment (toots) [inline src/generator/gen_unicode.ml:200]: Same

File: src/generator/gen_unicode.mli [hunk 3]
```diff
@@ -0,0 +1 @@
+(* empty on purpose *)
```
Comment (toots) [inline src/generator/gen_unicode.mli:1]: You can use `empty_module_interface_if_absent` from `dune` for that: https://dune.readthedocs.io/en/stable/dune-files.html#executable

**Review by toots — COMMENTED** [review]
This looks great too. I'll have another pass once the parent PR is merged.

File: src/generator/gen_unicode.ml (hunk 1 above)
Comment (hhugo) [reply src/generator/gen_unicode.ml:180]: Are you proposing to lift the constant at toplevel ? I'm not sure what you have in mind that would make it clear (` what list we're considering`)

File: src/generator/gen_unicode.ml (hunk 2 above)
Comment (hhugo) [reply src/generator/gen_unicode.ml]: done

File: src/generator/gen_unicode.ml (hunk 1 above)
Comment (hhugo) [reply src/generator/gen_unicode.ml]: done

**Comment by hhugo** [comment]
I've rebased this PR. ready for another round of review

**Review by toots — APPROVED** [review]
Looks great, thanks.

**Pull request merged**